thread_pthread.c

```diff
@@ -101,6 +101,78 @@
 #  endif
 #endif
 
+static gvl_hook_t * rb_gvl_hooks = NULL;
+static pthread_rwlock_t rb_gvl_hooks_rw_lock = PTHREAD_RWLOCK_INITIALIZER;
```

Comment:
I wonder whether we should use read-write locks or just plain locks. Read-write locks will probably improve performance slightly if there are GVL hooks, but it means that the hooks must be careful to ensure their code is thread-safe. Do we care so much about being as performant as possible when there are GVL hooks vs. the convenience of ensuring there can't be race conditions?